src/Event/Event.php

```diff
@@ -188,7 +188,7 @@ public function setResult($value = null)
      * Access the event data/payload.
      *
      * @param string|null $key The data payload element to return, or null to return all data.
-     * @return array|null The data payload if $key is null, or the data value for the given $key. If the $key does not
+     * @return mixed[]|mixed|null The data payload if $key is null, or the data value for the given $key. If the $key does not
```

Comment:
> Mixed means scalar

I don't think so. See http://php.net/manual/en/language.pseudo-types.php#language.types.mixed

What do you think what kind of array can not be passed `mixed[]`? Also, I am not sure how this annotation is related to PHP7.1. Are you talking about new nullable type such as `?string` ? Then I don't think it is related. Because `string` doesn't contain `null` apparently.